### PR TITLE
Saves the network type on dbase to prevent constant getCode checks

### PR DESCRIPTION
**Pull request by alexvandesande**
Tries to save the network type on dbase to prevent constant getCode checks, but for some reason the update is failing to be kept after a restart

### CODE REVIEW

**Comment by hiddentao** [comment]
So far watching the contract hasn't slowed the Wallet down to a crawl, so functionality-wise this works.

**Comment by alexvandesande** [comment]
@hiddentao can you check if it's saving the state on your database? The second time you run the app, tokens should open much faster as it won't do the check balance again.

@frozeman I'd appreciate if you tested this too:

```
_.pluck(CustomContracts.find().fetch(), 'network')
```

Switch networks, restart the app and see if the database keeps the network information correctly. If true it should keep track which tokens and contracts belong to the testnet, which belong in the main net, etc. Not working for me.

**Comment by hiddentao** [comment]
@alexvandesande Never had a problem with the speed at which token balance can be viewed. When you say "tokens should open much faster" which part of the UI are you referring to?

**Comment by alexvandesande** [comment]
In my machine it takes a few minutes before any contracts or tokens are
visible..

On Mon, May 30, 2016 at 6:10 AM, Ramesh Nair <notifications@github.com>
wrote:

> @alexvandesande <https://github.com/alexvandesande> Never had a problem
> with the speed at which token balance can be viewed. When you say "tokens
> should open much faster" which part of the UI are you referring to?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethereum/meteor-dapp-wallet/pull/196#issuecomment-222448668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAG5AizXssAIF4M5J_z83ZpNKN6KEjmqks5qGqmFgaJpZM4Ij4OF>
> .
>



-- 

Alex Van de Sande
UX Designer

**Comment by hiddentao** [comment]
I definitely don’t have that problem.

-- 
Ramesh Nair

Director
HiddenTao Ltd (UK company no. 6807289)

http://hiddentao.com
https://github.com/hiddentao
https://www.linkedin.com/in/hiddentao


On 30 May 2016 at 22:09:42, Alex Van de Sande (notifications@github.com) wrote:

In my machine it takes a few minutes before any contracts or tokens are
visible..

On Mon, May 30, 2016 at 6:10 AM, Ramesh Nair <notifications@github.com>
wrote:

> @alexvandesande <https://github.com/alexvandesande> Never had a problem
> with the speed at which token balance can be viewed. When you say "tokens
> should open much faster" which part of the UI are you referring to?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethereum/meteor-dapp-wallet/pull/196#issuecomment-222448668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAG5AizXssAIF4M5J_z83ZpNKN6KEjmqks5qGqmFgaJpZM4Ij4OF>
> .
>



--

Alex Van de Sande
UX Designer

**Comment by luclu** [comment]
It's the same for me @alexvandesande.

File: app/client/lib/ethereum/observeEvents.js [hunk 1]
```diff
@@ -21,26 +21,26 @@ observeEvents = function(){
         added: function(newDocument) {
             
             // This creates a temporary cache for the contracts, to reduce the amount of db read and writes            
-            var customContract = customContractsCache[newDocument.address.toLowerCase()];
+            // var customContract = customContractsCache[newDocument.address.toLowerCase()];
```
Comment (frozeman) [inline app/client/lib/ethereum/observeEvents.js]: Why are you commenting this out here?

Comment (alexvandesande) [inline app/client/lib/ethereum/observeEvents.js]: Should not be, will uncomment it

**Comment by alexvandesande** [comment]
I moved the token balance part in a separate PR as I'm not sure this code actually results in improved performance
https://github.com/ethereum/meteor-dapp-wallet/pull/206

**Pull request closed without merging**